From a7739ff1e551ad1386557550cff5073cdb0aee66 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:05 -0400
Subject: [ethernet] i40evf: Enable the ndo_set_features netdev op

Message-id: <1409663591-32531-23-git-send-email-sassmann@redhat.com>
Patchwork-id: 90499
O-Subject: [RHEL7.1 PATCH 022/148] i40evf: Enable the ndo_set_features netdev op
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Greg Rose <gregory.v.rose@intel.com>

Set netdev->hw_features to enable the ndo_set_features netdev op.

Change-Id: I5a086fbfa5a089de5adba2800c4d0b3a73747b11
Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 3415e8ce0de0242811b1ce9b89cdd8166a505959)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index b2c03bc..3d3ab14 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -2036,6 +2036,7 @@ static void i40evf_init_task(struct work_struct *work)
        NETIF_F_IPV6_CSUM |
        NETIF_F_TSO |
        NETIF_F_TSO6 |
+       NETIF_F_RXCSUM |
        NETIF_F_GRO;
 
  if (adapter->vf_res->vf_offload_flags
@@ -2046,6 +2047,10 @@ static void i40evf_init_task(struct work_struct *work)
         NETIF_F_HW_VLAN_CTAG_FILTER;
  }
 
+ /* copy netdev features into list of user selectable features */
+ netdev->hw_features |= netdev->features;
+ netdev->hw_features &= ~NETIF_F_RXCSUM;
+
  if (!is_valid_ether_addr(adapter->hw.mac.addr)) {
   dev_info(&pdev->dev, "Invalid MAC address %pMAC, using random\n",
     adapter->hw.mac.addr);
-- 
1.7.1